.github/workflows/yamllint.yml

```diff
@@ -0,0 +1,21 @@
+---
+name: yamllint
+'on':
+  pull_request:
+    types:
+      - opened
+jobs:
+  yamllint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+      - name: Install yamllint
+        run: pip install yamllint
```

Comment:
Is there no Go equivalent we can use?

If we're gonna be using Python-based tools for a Golang repo, migt as well use [pre-commit][https://pre-commit.com/] and enjoy its extensive ecosystem and the ability to also run locally.

Reply:
I didn't find anything Go-equivalent, almost anything is ending up using the same yamllint tool. we might use [this](https://golangci-lint.run/) to lint go itself.

There are a lot of python-based tools we can use (like pre-commit you mentioned), I'm not convinced it is good to limit ourselves to golang tools only because the repo itself uses golang.

Reply:
Adding a Python-based tool turns Python (the interpreter) into a dependency, and also forces us to consider how to manage the tools and packages we want to use.

So again, if we must, lets use `pre-commit` which has a rather elegant solution for the problem of installing Python tools and also allows us to run all the tests locally as well as in CI.

Reply:
Will do
as for Yaml checks, I see we have hundreds of errors, mostly in the tekton/deployment yamls, is there any automatic formatter for yamls we wish to use, or will we ignore those?

Reply:
most of them are indentation and long lines,
fixing them manually will take hours, that's why I'd prefer a formatter.

Reply:
I don't know of one, maybe you can make VScode help you

Reply:
formatted with YAML sort and some manual changes.
also added a rule that allow spaces in braces (which we are using in templates)